odoo/orm/domains.py

```diff
@@ -401,10 +401,20 @@ def map_conditions(self, function: Callable[[DomainCondition], Domain]) -> Domai
         """Map a function to each condition and return the combined result"""
         return self
 
-    def validate(self, model: BaseModel) -> None:
+    def validate(self, model: BaseModel, *, in_db: bool = False) -> None:
         """Validates that the current domain is correct or raises an exception"""
-        # just execute the optimization code that goes through all the fields
-        self._optimize(model, OptimizationLevel.FULL)
+        try:
+            query = model._search(self, active_test=False, bypass_access=True)
```

Comment:
Why bypassing the security here ? To avoid the costly override ? Because we can want to test validate the domain with a specific user, right ?

Maybe we can use limit=1 to mitigate the override cost?

Reply:
We just want to validate if the domain is correct. We don't need to check if the security domain is also correct.

What do you mean by limit=1? We just ask the explain plan, so limits do not matter.

Reply:
> We just want to validate if the domain is correct.

But it might be incorrect because of the ir.rule added somehow or depending of the user :thinking: (e.g. a search method) ? I am not sure if it matters, just asking ?

> What do you mean by limit=1? We just ask the explain plan, so limits do not matter.

I assume that you did want to bypass override of `_search` with `bypass_access=True`, that's why I proposed add limit=1 to avoid making _search override searching all records.